app/components/ui/create-tale-modal/component.js

```diff
@@ -0,0 +1,158 @@
+import Ember from 'ember';
+import Component from '@ember/component';
+import {inject as service} from '@ember/service';
+import {later} from '@ember/runloop';
+
+export default Component.extend({
+  store: service(),
+  router: service(),
+  
+  title: "",
+  imageId: "",
+  dataSet: [],
+  config: {},
+
+  createAndLaunch: true,
+  createButtonText: Ember.computed('createAndLaunch', function() {
+    return this.get('createAndLaunch') ? 'Create New Tale and Launch' : 'Create New Tale';
+  }),
+
+  disabled: Ember.computed('title', 'imageId', function() {
+    return (this.get('title') && this.get('imageId')) ? '' : 'disabled';
+  }),
+  
+  defaultErrorMessage: "There was an error while creating your Tale.",
+
+  // ---------------------------------------------------------------------------------
+  // ACTIONS TOC:
+  //   createNewTaleButton() = Event: onClick "Create New Tale" button
+  //   clearModal()          = Event: onHide modal 
+  //   setModalFromQueryParams() = Event: onShow modal
+  //   setDefaultAction()    = Event: onSelect "Create New Tale" dropdown
+  // ---------------------------------------------------------------------------------
+  actions: {
+    // ---------------------------------------------------------------------------------
+    // Either imports tale from a dataset or creates a new tale
+    // ---------------------------------------------------------------------------------
+    async createNewTaleButton() {
```

Comment:
I don't think `async` is needed here, since we aren't calling `await` and neither of the functions called below will return a Promise.

Reply:
Correct. It used to have code in there but was split up into two different functions.